packages/babel-core/src/config/files/plugins.ts

```diff
@@ -122,22 +115,26 @@ function* resolveAlternativesHelper(
 }
 
 function tryRequireResolve(
-  id: Parameters<RequireResolve>[0],
-  { paths: [dirname] }: Parameters<RequireResolve>[1],
+  id: string,
+  dirname: string | undefined,
 ): Result<string> {
   try {
-    return { error: null, value: require.resolve(id, { paths: [dirname] }) };
+    if (dirname) {
+      return { error: null, value: require.resolve(id, { paths: [dirname] }) };
+    } else {
+      return { error: null, value: require.resolve(id) };
```

Comment:
This change is necessary because sometimes `dirname` is undefined here, and Node.js 20 as stricter validation.